tooling/templatize/main.go

```diff
@@ -1,24 +1,81 @@
 package main
 
 import (
+	"fmt"
 	"log"
+	"os"
 
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/Azure/ARO-HCP/tooling/templatize/config"
 )
 
 func DefaultGenerationOptions() *GenerationOptions {
 	return &GenerationOptions{}
 }
 
 type GenerationOptions struct {
-	Input  string
-	Region string
-	User   string
+	ConfigFile string
+	Input      string
+	Cloud      string
+	DeployEnv  string
+	Region     string
+	User       string
+}
+
+func (opts *GenerationOptions) Validate() error {
```

Comment:
Can be made more simple using cobra

Reply:
i covered the basic validations with cobra

Reply:
I also know this is past, but a general comment from my experience: any CLI worth something is two things - a tiny wrapper of code to get input parameters from a command line, and a useful utility library on the inside. The utility library is where the value lies, and the moment it gets re-used, whether that's someone writing new automations that want to call the library as opposed to shelling out, or it's some CLI that embeds part of this one, that's the moment you can't use Cobra to validate your options. Your `func (opts *Options) Run()` needs to be certain about the content of `*Options` and if someone's creating that in Go code with a struct literal, your Cobra code won't run :)